test/parallel_api/algorithm/alg.sorting/sort_by_key_common.h

```diff
@@ -0,0 +1,289 @@
+void
+test_with_std_policy(Policy&& policy, Size n, Compare compare, StabilityTag stability_tag)
+{
+    std::vector<KeyT> origin_keys(n);
+    std::vector<ValT> origin_vals(n);
+    generate_data(origin_keys.data(), origin_vals.data(), n, 42);
+    std::vector<KeyT> keys(origin_keys);
+    std::vector<ValT> vals(origin_vals);
+
+    call_sort(policy, keys.begin(), vals.begin(), n, compare, stability_tag);
+    check_sort(keys.begin(), vals.begin(), origin_keys.begin(), origin_vals.begin(), n, compare, stability_tag);
+
+}
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <typename KeyT, typename ValT, sycl::usm::alloc alloc_type, std::uint32_t KernelNameID,
+          typename Size, typename Compare, typename StabilityTag>
+void
+test_with_usm(sycl::queue& q, Size n, Compare compare, StabilityTag stability_tag)
+{
+    std::vector<KeyT> origin_keys(n);
+    std::vector<ValT> origin_vals(n);
+    generate_data(origin_keys.data(), origin_vals.data(), n, 42);
+    std::vector<KeyT> keys(origin_keys);
+    std::vector<ValT> vals(origin_vals);
+    TestUtils::usm_data_transfer<alloc_type, KeyT> keys_device(q, keys.begin(), keys.end());
+    TestUtils::usm_data_transfer<alloc_type, ValT> vals_device(q, vals.begin(), vals.end());
+
+    // calling sort
+    auto policy = TestUtils::make_device_policy<TestUtils::unique_kernel_name<class USM, KernelNameID>>(q);
+    call_sort(policy, keys_device.get_data(), vals_device.get_data(), n, compare, stability_tag);
+
+    // checking results
+    keys_device.retrieve_data(keys.begin());
+    vals_device.retrieve_data(vals.begin());
+   // sort_by_key with device policy guarantees stability, hence StableSort{} is passed
+    check_sort(keys.begin(), vals.begin(), origin_keys.begin(), origin_vals.begin(), n, compare, StableSort{});
+}
+
+template <typename KeyT, typename ValT, std::uint32_t KernelNameID,
+          typename Size, typename Compare, typename StabilityTag>
+void
+test_with_buffers(sycl::queue& q, Size n, Compare compare, StabilityTag stability_tag)
+{
+    std::vector<KeyT> origin_keys(n);
+    std::vector<ValT> origin_vals(n);
+    generate_data(origin_keys.data(), origin_vals.data(), n, 42);
+    std::vector<KeyT> keys(origin_keys);
+    std::vector<ValT> vals(origin_vals);
+
+    // calling sort
+    {
+        sycl::buffer<KeyT> keys_device(keys.data(), n);
+        sycl::buffer<ValT> vals_device(vals.data(), n);
+        auto policy = TestUtils::make_device_policy<TestUtils::unique_kernel_name<class Buffer, KernelNameID>>(q);
+        call_sort(policy, oneapi::dpl::begin(keys_device), oneapi::dpl::begin(vals_device), n, compare, stability_tag);
+    }
+   // sort_by_key with device policy guarantees stability, hence StableSort{} is passed
+    check_sort(keys.begin(), vals.begin(), origin_keys.begin(), origin_vals.begin(), n, compare, StableSort{});
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+#if TEST_DPCPP_BACKEND_PRESENT
```

Comment:
```suggestion

```
Looks like these two lines simple not required.

Reply:
Removed